### PR TITLE
Fix the JNI error 'global reference table overflow' caused by receiving a large number of WebSocket messages in the background

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/cocos-engine/issues/17552
### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/network/WebSocket-okhttp.cpp [hunk 1]
```diff
@@ -372,7 +360,7 @@ JNI_PATH(nativeOnStringMessage)(JNIEnv * /*env*/,
                                 jlong handler) {
     auto *wsOkHttp3 = HANDLE_TO_WS_OKHTTP3(handler); // NOLINT(performance-no-int-to-ptr)
     std::string msgStr = JniHelper::jstring2string(msg);
-    RUN_IN_GAMETHREAD(wsOkHttp3->onStringMessage(msgStr));
+    wsOkHttp3->onStringMessage(msgStr);
```
Comment (minggo) [inline native/cocos/network/WebSocket-okhttp.cpp:363]: Why change it?

Comment (bofeng-song) [reply native/cocos/network/WebSocket-okhttp.cpp:363]: > Why change it?

The Java side has ensured that nativeOnStringMessage runs on the game thread.

Comment (minggo) [reply native/cocos/network/WebSocket-okhttp.cpp:363]: I see. But i don't know why need to do synchronization in java instead in c++?

Comment (bofeng-song) [reply native/cocos/network/WebSocket-okhttp.cpp:363]: While CocosActivity run on background, the old logic will create jni's global object. And the number of global object may exeed system's limit.

Comment (minggo) [reply native/cocos/network/WebSocket-okhttp.cpp:363]: The current implementation also use jni to invoke java codes. Why it doesn't have the problem?

Comment (bofeng-song) [reply native/cocos/network/WebSocket-okhttp.cpp:363]: The global object is created during the nativeOnBinaryMessage call.

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**